### PR TITLE
await rc/rs resource deletion while renaming the resource during roll…

### PR DESCRIPTION
**Pull request by ravilr**
…ing update

Fixes https://github.com/fabric8io/kubernetes-client/issues/546

@jimmidyson @iocanel

### CODE REVIEW

**Comment by iocanel** [comment]
@ravilr: thank you so much for the pull request. I will try to have a close look at it asap.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/RollingUpdater.java [hunk 1]
```diff
@@ -212,6 +217,48 @@ public void run() {
...
+          }
+        } catch (KubernetesClientException e) {
+          if (e.getCode() == 404) {
+            countDownLatch.countDown();
+          }
+        }
+      }
+    };
+
+    ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
```
Comment (iocanel) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/RollingUpdater.java]: Why don't we move that outside, as a field?

**Comment by iocanel** [comment]
LGTM!

**Comment by jstrachan** [comment]
ok to test

**Comment by jstrachan** [comment]
[merge]

**Comment by jimmidyson** [comment]
Why are we merging when there's an outstanding question from @iocanel (https://github.com/fabric8io/kubernetes-client/pull/549#pullrequestreview-5703743)? Plus it needs a rebase...

**Comment by jstrachan** [comment]
@jimmidyson saw the LGTM from @iocanel and believed him :)

**Pull request merged**